### PR TITLE
test(program): Add unit tests for statistics.py (0% -> full coverage)

### LINKED ISSUE #4366: Add unit tests for statistics.py (0% coverage)
## Problem Statement

`esp/esp/program/statistics.py` currently has **0% test coverage**. This issue tracks adding a comprehensive test suite for all public statistics functions.

## What this covers

The module has 10 public functions, all testable as pure Python:

- `zipcodes` — zip code aggregation with limit support
- `demographics` — grade/age/financial aid stats
- `schools` — k12school vs plain school counts
- `startreg` — first registration and confirmation dates per program
- `repeats` — cross-program repeat student counts
- `heardabout` — normalised "heard about" reasons
- `hours` — enrolled/attended timeslot binning
- `student_reg` — lottery/enrol/check-in counts
- `teacher_reg` — teacher registration stage counts
- `class_reg` — class count and class-student-hours

## Test design

- Uses `ProgramFrameworkTest` as base (matches project convention)
- Tests verify both the return type (string) and `result_dict` key/value correctness
- Edge cases: empty querysets, limit=0, limit=1, zero financial aid

Part of #3780

### PR DESCRIPTION
**Pull request by moksha-hub**
Adds comprehensive unit tests for `esp/esp/program/statistics.py`, which previously had **0% test coverage**.

Closes #4366
Part of #3780

## Changes

- New file: `esp/esp/program/test_statistics.py`
- **10 test classes, 38 tests** covering all public functions

## Test classes

| Class | Function tested |
|---|---|
| `ZipcodesTest` | `zipcodes` |
| `DemographicsTest` | `demographics` |
| `SchoolsTest` | `schools` |
| `StartregTest` | `startreg` |
| `RepeatsTest` | `repeats` |
| `HeardaboutTest` | `heardabout` |
| `HoursTest` | `hours` |
| `StudentRegStatTest` | `student_reg` |
| `TeacherRegTest` | `teacher_reg` |
| `ClassRegTest` | `class_reg` |

### CODE REVIEW

**Comment by moksha-hub** [comment]
@Oval17 new issues found by me.
Please have a review.
Thanks a lot. for your support.

**Comment by Oval17** [comment]
Will review soon @moksha-hub

File: esp/esp/program/test_statistics.py [hunk 1]
```diff
@@ -0,0 +1,512 @@
...
+complete program fixture including teachers, scheduled class sections, and
+students.  We convert the list attributes to QuerySets where functions require
+them.
+"""
+
+from types import SimpleNamespace
+
+from django.test import TestCase
+
+from esp.program.models import Program, StudentRegistration, RegistrationType
```
Comment (Oval17) [inline esp/esp/program/test_statistics.py]: Kindly see for unused imports and remove them!!

File: esp/esp/program/test_statistics.py [hunk 2]
```diff
@@ -0,0 +1,512 @@
...
+
+    def setUp(self, *args, **kwargs):
+        # Smaller fixture so tests run faster.
+        kwargs.setdefault("num_timeslots", 2)
+        kwargs.setdefault("timeslot_length", 50)
+        kwargs.setdefault("timeslot_gap", 10)
+        kwargs.setdefault("num_teachers", 3)
+        kwargs.setdefault("classes_per_teacher", 1)
+        kwargs.setdefault("sections_per_class", 1)
+        kwargs.setdefault("num_rooms", 3)
```
Comment (Oval17) [inline esp/esp/program/test_statistics.py]: Without add_user_profiles() being called first, 
no RegistrationProfile rows exist in the DB, so the .filter() on lines 71 and 73 returns empty QuerySets and any test that iterates over self.student_profiles or self.teacher_profiles silently passes without exercising any logic.
tests for zipcodes, demographics, schools, heardabout etc pass vacuously 
Kindly see this

**Comment by moksha-hub** [comment]
@willgearty  i would like to have a review as it says "Copilot encountered an error and was unable to review this pull request. You can try again by re-requesting a review."

**Comment by willgearty** [comment]
One last failing test.
